File: benchmarks/compute-pi/pi.pl
use strict;
use warnings;

my $n = 10000000;
my $step = 1.0 / $n;
my $sum = 0.0;

my $i = 0;
while ($i < $n) {
    my $x = ($i + 0.5) * $step;
    $sum += 4.0 / (1.0 + $x * $x);
    $i = $i + 1;
}

my $pi = $sum * $step;
print "$pi\n";

File: benchmarks/compute-pi/pi.js
const n = 10_000_000
const step = 1.0 / n
let sum = 0.0

for (let i = 0; i < n; i++) {
	const x = (i + 0.5) * step
	sum += 4.0 / (1.0 + x * x)
}

const pi = sum * step
console.log(pi)
